### ISSUE TITLE
Error when importing nimbusml 1.0.0

### ISSUE THREAD

**Issue by pieths** [microsoft/NimbusML#111]
Create a fresh python 3.7.3 virtual environment (`python -m venv [destDir]`) and pip install:

- nimbusml 1.0.0 (from local build)
- jupyterlab
- matplotlib

Then start `jupyter lab`. Enter the following in the first cell of a new notebook,

```python
import numpy as np
import pandas as pd
import nimbusml as nml
```

Running the cell prints out the following error message.

```python
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-3-27d67527a163> in <module>
      1 import numpy as np
      2 import pandas as pd
----> 3 import nimbusml as nml

c:\nimbusml\venv\lib\site-packages\nimbusml\__init__.py in <module>
     23 from .internal.utils.data_stream import FileDataStream
     24 from .internal.utils.utils import run_tests
---> 25 from .pipeline import Pipeline
     26 
     27 if sys.platform.lower() == "linux":

c:\nimbusml\venv\lib\site-packages\nimbusml\pipeline.py in <module>
     19 from sklearn.utils.validation import check_X_y, check_array
     20 
---> 21 from .internal.core.base_pipeline_item import BasePipelineItem
     22 from .internal.entrypoints.data_customtextloader import \
     23     data_customtextloader

c:\nimbusml\venv\lib\site-packages\nimbusml\internal\core\base_pipeline_item.py in <module>
     19 
     20 import six
---> 21 from sklearn.utils.fixes import signature
     22 
     23 from ..utils.data_roles import DataRoles, Role

ImportError: cannot import name 'signature' from 'sklearn.utils.fixes' (c:\nimbusml\venv\lib\site-packages\sklearn\utils\fixes.py)
```

**Comment by pieths** [comment]
It looks like this might be a problem with scikit-learn 0.21.*. Downgrading to version 0.20.* gets rid of the error.

**Comment by pieths** [comment]
*Note: the same error will also occur when executing the `Component.get_params(...)` method in `internal\utils\entrypoints.py`.*

The `signature` method in `sklearn.utils.fixes` was removed in version 0.21.rc2.
See [this issue](https://github.com/scikit-learn/scikit-learn/issues/13388) for more details.

The `signature` method exists in `fixes.py` because of the following code.

```python
try:
    from inspect import signature
except ImportError:
    from ..externals.funcsigs import signature
```

Though the `inspect` module exists in both Python 2.x and 3.x, the `inspect.signature` method is new in Python 3.3.

The `externals.funcsigs` module is a

> Back port of Python 3.3's function signature tools from the inspect
> module, modified to be compatible with Python 2.6, 2.7 and 3.2+.

`funcsigs.py` and its related files in `sklearn.externals` were copied in to the scikit-learn source tree from the [funcsigs module](https://pypi.org/project/funcsigs/).

Here are some options to fix this issue:

1. Remove support for Python < 3.3. The code could then just use the inspect module from the standard library.

2. Limit the version of the scikit-learn dependency in NimbusML to [0.19.0, 0.20.3].

3. Add the [funcsigs](https://pypi.org/project/funcsigs/) package as a dependency of NimbusML.

4. Add the following code to both files that require it. This has the drawback that it relies on a non-public api in scikit-learn.

    ```python
    try:
        from inspect import signature
    except ImportError:
        from sklearn.externals.funcsigs import signature
    ```

5. Import the `funcsigs` code in to NimbusML (like scikit-learn does).

**Comment by ganik** [comment]
Thank you @pieths for detailed investigation. I suggest to go with option 4 and rely on scikit-learn to deal with signature. Option 1 is no good, we don't want to cut python 2.7 bcs of this minor thing. Same goes with scikit learn version, this would force users to downgrade from latest scikit-learn version. Adding another dependency or copying funcsigs are inferior to option 4.
